rmf_task/include/rmf_task/BinaryPriorityScheme.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_TASK__BINARYPRIORITYSCHEME_HPP
+#define RMF_TASK__BINARYPRIORITYSCHEME_HPP
+
+#include <memory>
+
+namespace rmf_task {
+
+// Forward declare abstract interfaces. The definition will remain internal
```

Comment:
I recommend putting these forward declarations and type aliases into their own headers. Probably a `Priority.hpp` and `CostCalculator.hpp`.

I realize it probably seems unnecessary for some plain forward declarations, but it's best to have a single source of truth for the names of these classes.

Reply:
Done https://github.com/osrf/rmf_core/pull/278/commits/cced88c75d76dbcef2d39268700d388cca86146c